filters/bloom_filter.go

```diff
@@ -0,0 +1,600 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package filters provides probabilistic membership data structures for efficient
+// set membership testing with controlled false positive rates.
+//
+// The Bloom filter is a space-efficient probabilistic data structure that is used
+// to test whether an element is a member of a set. False positive matches are
+// possible, but false negatives are not. This implementation uses XXHash64 with
+// Kirsch-Mitzenmacher double hashing optimization.
+package filters
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"math/bits"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+// BloomFilter is a probabilistic data structure for set membership testing.
+// It provides constant-time updates and queries with a configurable false
+// positive rate. No false negatives are possible.
+type BloomFilter interface {
```

Comment:
matter of tastes problem: usually "Get" prefix doesn't recommended in Getter. ref: https://go.dev/doc/effective_go#Getters